lambda/script.py

```diff
@@ -0,0 +1,154 @@
+import json
+import os
+from typing import Dict, List, Optional
+
+import boto3  # type: ignore
+import googleapiclient.discovery  # type: ignore
+from google.oauth2 import service_account  # type: ignore
+
+
+def get_env_var(env) -> Optional[str]:
+    """
+    Returns an enviroment variable.
+    """
+    return os.environ.get(env)
+
+
+def get_scope(scope: Optional[str]) -> List[str]:
+    """
+    Returns the scope from AWS SSM.
+    """
+    client = boto3.client("ssm")
+    return [client.get_parameter(Name=scope, WithDecryption=True)["Parameter"]["Value"]]
+
+
+def get_subject_email(subject: Optional[str]) -> str:
+    """
+    Returns the subject email from AWS SSM
+    """
+    client = boto3.client("ssm")
+    return client.get_parameter(Name=subject, WithDecryption=True)["Parameter"]["Value"]
+
+
+def get_credentials_file(credentials: Optional[str]) -> str:
+    """
+    Gets the credentials file from AWS SSM and writes it to the local file /tmp/credentials.json
+    Returns the file location for the credentials file.
+    """
+    client = boto3.client("ssm")
+    content = client.get_parameter(Name=credentials, WithDecryption=True)["Parameter"][
+        "Value"
+    ]
+
+    with open("/tmp/credentials.json", "w") as outfile:
+        outfile.write(content)
+    return "/tmp/credentials.json"
+
+
+def create_client(
+    api: str,
+    api_version: str,
+    service_account_file: str,
+    scope: List[str],
+    subject: str,
+    pageToken: str = None,
+):
+    """
+    Returns a given client built from a google api and api version passed in .
+    """
+    client = boto3.client("ssm")
+    credentials = service_account.Credentials.from_service_account_file(
+        service_account_file, scopes=scope, subject=subject,
+    )
+
+    client = googleapiclient.discovery.build(
+        api, api_version, credentials=credentials, cache_discovery=False
+    )
+
+    return client
+
+
+def build_group_dict(api, api_version, scope) -> Dict[str, str]:
```

Comment:
Quite a few of these functions aren't typing their parameters. Even if they're not checked by mypy due to boto3 ignoring, it's helpful documentation

Reply:
All function parameters are now typed.